### DISCUSSION TITLE
Button hoverStyle props missing

### DISCUSSION THREAD

**Discussion by ALegendsTale** [tamagui/tamagui#247] [General]
**Describe the bug**
Button extends other components with additional props as per the [docs](https://tamagui.dev/docs/components/button/1.0.0-beta.161). Some of those additional props aren't available on `hoverStyle` though (e.g. color, fontWeight, etc)

**To Reproduce**
Add a button with the hoverStyle prop & attempt to change text color on hover. Typescript will warn the prop isn't available on hoverStyle & attempting to compile won't produce any results.
```
<Button
     color='blue'
     hoverStyle={{
          color: 'red',     // Error missing prop on hoverStyle
     }}
>
     Text
</Button>
```

**Comment by natew** [comment]
You can pass a separate ButtonText to Button but unfortunately we can’t really do hover > child yet properly as it doesn’t exist in React Native. 

We could support it, I have some ideas for it in next.md

**Comment by natew** [comment]
Moving to discussion as it's not broken per se.

**Comment by VladyslavKochetkov** [comment]
+1 from me for this idea. We kind of got around it with the following. Our use case is a button with multiple variants (primary, alternative and text) where for primary and alternative the color stays white, but on text changes on hover.

```
color: '$white',
textProps: {
  hoverStyle: {
    color: '$purple400',
  },
  pressStyle: {
    color: '$purple400',
  },
  focusStyle: {
    color: '$purple400',
  },
},
hoverStyle: {
  backgroundColor: 'transparent',
  borderColor: 'transparent',
},
pressStyle: {
  backgroundColor: '$purple600Dark',
  borderColor: 'transparent',
},
focusStyle: {
  borderColor: '$purple200Dark',
},
```

But this specifically only triggers when hovering the text, not the button, and also doesn't seem to forward to color prop to the icon.

**Comment by jonsherrard** [comment]
Any other thoughts on hover > child? 

Does this update to gesture handler help things at all?: https://twitter.com/breskin67/status/1646511002933153794
